Shipr-Helm/templates/redis-deployment.yaml

```diff
@@ -0,0 +1,72 @@
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: {{ Values.redis.name }}-pv
+spec:
+  capacity:
+    storage: {{ Values.redis.volume.storage }}
+  accessModes:
+    - {{ Values.redis.volume.accessModes }}
+  hostPath:
+    path: {{ Values.redis.volume.hostPath }}
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: {{ Values.redis.name }}-pvc
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: {{ Values.redis.storage }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ Values.redis.name }}
+spec:
+  replicas: {{ Values.redis.replicas }}
+  selector:
+    matchLabels:
+      app: {{ Values.redis.name }}
+  template:
+    metadata:
+      labels:
+        app: {{ Values.redis.name }}
+    spec:
+      containers:
+      - name: {{ Values.redis.name }}
+        image: {{ Values.redis.repository }}:{{ Values.redis.tag }}
+        ports:
+        - containerPort: {{ Values.redis.containerPort }}
+        livenessProbe:
+          tcpSocket:
+            port: {{ Values.redis.containerPort }}
+          initialDelaySeconds: 10
+          periodSeconds: 10
+        readinessProbe:
+          tcpSocket:
+            port: {{ Values.redis.containerPort }}
+          initialDelaySeconds: 5
+          periodSeconds: 5
+        volumeMounts:
+        - mountPath: /data
+          name: redis-storage
+      volumes:
+      - name: redis-storage
+        persistentVolumeClaim:
+          claimName: redis-pvc
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ Values.redis.name }}
+spec:
+  selector:
+    app: {{ Values.redis.name }}
+  ports:
+    - protocol: TCP
+      port: {{ Values.redis.service.port }}
+      targetPort: {{ Values.redis.service.port }}
+  type: ClusterIP
```

Comment:
Hard coding service type? What if users want the redis service to be of loadbalancer type?

Reply:
Used dynamic value